.github/workflows/upload.yaml

```diff
@@ -0,0 +1,31 @@
+name: Upload charm
+
+on:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    name: Charmhub upload
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Check libraries
+        uses: canonical/charming-actions/check-libraries@1.0.2
+        with:
+          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Select charmhub channel
+        uses: canonical/charming-actions/channel@1.0.2
+        id: channel
+
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@1.0.2
+        with:
+          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
```

Comment:
We should add the github secret for CHARMCRAFT_AUTH

Reply:
Yes.
Added.